src/poprox_recommender/training/train.py

```diff
@@ -0,0 +1,125 @@
+
+import torch
+from safetensors.torch import load_file
+from transformers import Trainer, TrainingArguments
+
+from poprox_recommender.evaluation import evaluate
+from poprox_recommender.paths import project_root
+from poprox_recommender.training.dataset import BaseDataset, ValDataset
+
+logger = logging.getLogger(__name__)
+root = project_root()
+
+
+def train(device, load_checkpoint):
+    """
+    1. Initialize model
+    """
+
+    logger.info("Initialize Model")
+    Model = getattr(importlib.import_module("poprox_recommender.model.nrms"), "NRMS")
+    model = Model(args).to(device)
+
+    if load_checkpoint:
+        checkpoint = load_file(args.checkpoint_path)
+        model.load_state_dict(checkpoint)
+
+    """
+    def print_model_size(model):
+        print("Model: ", model.__class__.__name__)
+        total_params = 0
+        for name, param in model.named_parameters():
+            print(f"Layer: {name} | Size: {param.size()} | Number of parameters: {param.numel()}")
+            total_params += param.numel()
+        print(f"Total number of parameters in the model: {total_params}")
+    """
+
+    """
+    2. Load Data & Create Dataset
+    """
+    logger.info("Initialize Dataset")
+    # train_dataset = root / "data/MINDlarge_post_train/behaviors_parsed.tsv"
+    train_dataset = BaseDataset(
+        args,
+        root / "data/MINDlarge_post_train/behaviors_parsed.tsv",
+        root / "data/MINDlarge_post_train/news_parsed.tsv",
+    )
+    # eval_dataset = root / "data/MINDlarge_dev/behaviors.tsv"
+    eval_dataset = ValDataset(args, root / "data/MINDlarge_dev/behaviors.tsv", root / "data/MINDlarge_dev/news.tsv")
+
+    """
+    3. Train
+    """
+    logger.info("Training Start")
+    training_args = TrainingArguments(
+        output_dir="models/nrms-mind",
+        logging_strategy="steps",
+        save_total_limit=5,
+        lr_scheduler_type="constant",
+        weight_decay=0.0,
+        optim="adamw_torch",
+        save_strategy="epoch",
+        eval_strategy="no",
+        gradient_accumulation_steps=8,
+        learning_rate=1e-4,
+        per_device_train_batch_size=16,
+        per_device_eval_batch_size=1,
+        num_train_epochs=3,
+        remove_unused_columns=False,
+        logging_dir="models/nrms-mind",
+        logging_steps=1,
+        report_to=None,
+    )
+
+    trainer = Trainer(
+        model=model,
+        args=training_args,
+        train_dataset=train_dataset,
+        eval_dataset=eval_dataset,
+    )
+    trainer.train()
+
+    """
+    4. Evaluate model by validation dataset
+    """
+    logger.info("Evaluation")
+    auc, mrr, ndcg5, ndcg10 = evaluate(args, model, device)
+    logger.info(f"evaluation auc={auc}, mrr={mrr}, ndcg5={ndcg5}, ndcg10={ndcg10}")
+
+
+if __name__ == "__main__":
+    import os
+
+    os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
+    os.environ["CUDA_VISIBLE_DEVICES"] = "0"
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
```

Comment:
We need to remove this hard-coded device configuration, and use the POPROX default device instead.